src/main/docs/guide/httpServer/customArgumentBinding.adoc

```diff
@@ -0,0 +1,59 @@
+argument binder that supports that annotation.
+
+If no fitting annotation is found Micronaut will try to find an argument binder that supports the argument's type.
+
+An api:core.bind.ArgumentBinder[ArgumentBinder] can be registered as a bean. In this case it is only present when a
+api:context.BeanContext[BeanContext] is initialized.
+
+Binders are called by the same server logic that calls the api:io.micronaut.http.annotation.Controller[Controller]. Once
+the request is received and the argument is resolved in the api:http.bind.binders.ArgumentBinderRegistry[ArgumentBinderRegistry]
+api:io.micronaut.http.filter.HttpFilter[HttpFilter]s are called before returning results to the Controller. Your
+api:io.micronaut.core.bind.BindingResult[BindingResult] may depend on those being initialized before returning the binding
+result.
+
+A binding result of the class you are binding will be returned from the bind method in the implementation written. The
+result will be classified as either `SATISFIED` or `UNSATISFIED`. Returning a binding of the type `EMPTY` will be considered
+SATISFIED. If the returned binding is UNSATISFIED the registry will continue to attempt to find argument binders that match
+the argument.
+
+If the binding result throws an exception that is not handled it will bubble up to the api:io.micronaut.http.annotation.Controller[Controller]
+where it can be caught by an api:io.micronaut.http.annotation.Error[Error] handler.
+
+Key interfaces are:
+
+=== AnnotatedRequestArgumentBinders
+* api:http.bind.binders.AnnotatedRequestArgumentBinder[AnnotatedRequestArgumentBinder]
+
+- An interface for classes that bind an api:core.type.Argument[Argument] from an api:http.HttpRequest[HttpRequest] driven
+by an annotation.
+
+TIP: It is common to use api:io.micronaut.core.convert.ConversionService[ConversionService] to convert the data to the type of argument.
+
+=== TypedRequestArgumentBinders
+
+* api:http.bind.binders.TypedRequestArgumentBinder[TypedRequestArgumentBinder]
+
+- An interface for classes that bind an api:core.type.Argument[Argument] from an api:http.HttpRequest[HttpRequest] driven by
+the argument's type.
+
+snippet::io.micronaut.docs.http.bind.binders.MyBoundBeanTypedRequestArgumentBinder[tags="class", indent=0, title="Example of typed data binding"]
+<1> Override the bind method with the data type to bind, in this case MyBoundBean
+<2> After building the bound type as desired return the api:io.micronaut.core.bind.BindingResult[BindingResult]
+<3> Also override the argumentType method which is used by the ArgumentBinderRegistry.
+
+snippet::io.micronaut.docs.http.bind.binders.MyBoundBeanController[tags="typed", indent=0, title="A controller operation with this typed binding"]
+<1> The parameter will be bound with no annotation.
+
+snippet::io.micronaut.docs.http.bind.binders.MyBoundBeanAnnotatedRequestArgumentBinder[tags="class", indent=0, title="Example of annotated data binding"]
```

Comment:
Include a snippet of the annotation denoting that it is annotated with `@Bindable`